packages/api/src/providers/ActivitySendStatusTelemetry/ActivitySendStatusTelemetryComposer.tsx

```diff
@@ -0,0 +1,35 @@
+import { useGetActivityByKey, useSendStatusByActivityKey, useTrackEvent } from '../../hooks';
+
+import usePrevious from '../../hooks/internal/usePrevious';
+
+const ActivitySendStatusTelemetryComposer = () => {
+  const [activityToSendStatusMap] = useSendStatusByActivityKey();
+  const previousActivityToSendStatusMap = usePrevious(activityToSendStatusMap);
+  const getActivityByKey = useGetActivityByKey();
+  const trackEvent = useTrackEvent();
+
+  if (activityToSendStatusMap && previousActivityToSendStatusMap) {
+    const allActivityKeys = activityToSendStatusMap.keys();
+
+    for (const key of allActivityKeys) {
+      const currentStatus = activityToSendStatusMap.get(key);
+      const previousStatus = previousActivityToSendStatusMap.get(key);
+
+      if (!!currentStatus && !!previousStatus && currentStatus !== previousStatus) {
+        const activity = getActivityByKey(key);
+        const telemetryPayload = {
+          currentStatus: currentStatus.toString(),
+          previousStatus: previousStatus.toString(),
+          clientActivityID: activity?.channelData?.clientActivityID,
+          type: activity?.type?.toString(),
+          key,
+          hasAttachment: activity?.type === 'message' && activity?.attachments?.length > 0 ? 'true' : 'false'
+        };
+        trackEvent && trackEvent('send-status:change', telemetryPayload);

```

Comment:
Please correct me if I am wrong. I think `trackEvent` is never falsy.

Also consider eliminate one-use constants.

```suggestion
        trackEvent('send-status:change', telemetryPayload);
```

Reply:
updated, we need to use the const of trackEvent as useTrackEvent cannot directly used in a loop. 
![image](https://user-images.githubusercontent.com/55606459/218279635-bb5bf865-2871-40ea-bf09-f03ac125b598.png)